test/controllers/workshops_controller_test.rb

```diff
@@ -17,6 +17,24 @@ class WorkshopsControllerTest < ActionDispatch::IntegrationTest
     assert_select 'title', 'Hack Club Workshops'
   end
 
+  test 'renders root-level markdown files' do
+    get '/CONTRIBUTING.md'
+
+    # The page title should include file name and 'Hack Club Workshops'
+    assert_select 'title', 'CONTRIBUTING.md | Hack Club Workshops'
+
+    # There should be a sidebar
+    assert_select '.workshop-sidebar > ul > li > a'
+  end
+
+  test 'serves non-markdown files from root workshops directory' do
+    path = Rails.root.join('vendor', 'hackclub', 'workshops', 'test.txt')
+    FileUtils.touch(path)
+    get '/test.txt'
+    FileUtils.rm(path)
+    assert_response :success
+  end
```

Comment:
:thumbsup:

On Tue, Aug 16, 2016 at 1:12 PM, Max Wofford <notifications@github.com>
wrote:

> In test/controllers/workshops_controller_test.rb
> <https://github.com/hackclub/lecture-hall/pull/24#discussion_r75010449>:
>
> > +    get '/CONTRIBUTING.md'
> > +
> > +    # The page title should include file name and 'Hack Club Workshops'
> > +    assert_select 'title', 'CONTRIBUTING.md | Hack Club Workshops'
> > +
> > +    # There should be a sidebar
> > +    assert_select '.workshop-sidebar > ul > li > a'
> > +  end
> > +
> > +  test 'serves non-markdown files from root workshops directory' do
> > +    path = Rails.root.join('vendor', 'hackclub', 'workshops', 'test.txt')
> > +    FileUtils.touch(path)
> > +    get '/test.txt'
> > +    FileUtils.rm(path)
> > +    assert_response :success
> > +  end
>
> We originally decided to not support them, but later agreed on supporting
> them. I need to update the PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/hackclub/lecture-hall/pull/24/files/d2e0bd670711b0027e2182c9fd1765378d4c115c#r75010449>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA8j-Dw5i7isGseJOetHSOxwnX6SrGQSks5qghmzgaJpZM4JhzUZ>
> .
>



-- 
Zach Latta
https://zachlatta.com <http://zachlatta.com>